gateway/src/worker.rs

```diff
@@ -57,7 +61,16 @@ impl Worker<BoxedTask> {
 
         while let Some(mut work) = self.recv.recv().await {
             loop {
-                match work.poll(()).await {
+                let timeout = time::sleep(Duration::from_secs(TASK_MAX_IDLE_TIMEOUT));
+                let mut poll = work.poll(());
+                let res = tokio::select! {
+                    res = &mut poll => res,
+                    _ = timeout => {
+                        warn!("a task has been running for a long time");
```

Comment:
Is it possible to log the project that timed out?

Reply:
Ah, yes, absolutely. Coming up.